presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcessFactory.java

```diff
@@ -66,11 +68,14 @@ public NativeExecutionProcessFactory(
         this.workerProperty = requireNonNull(workerProperty, "workerProperty is null");
     }
 
-    public NativeExecutionProcess createNativeExecutionProcess(
+    public synchronized NativeExecutionProcess createNativeExecutionProcess(
```

Comment:
Do you think the name of the function is misleading after this change ? Maybe rename this to `getNativeExecutionProcess()` ?